doc/glossary.rst

```diff
@@ -1848,26 +1848,72 @@ See concept :term:`sample property`.
+        A relative weight for each sample. Intuitively, if all weights are
+        integers, using them in a model or scorer is like duplicating each
+        sample as many times as the weight value. Weights may be specified as
+        floats, so that sample weights are usually equivalent up to a constant
+        positive scaling factor.
+
+        `sample_weight` can be both an argument of the estimator's `fit` method
+        for model training or a parameter of a :term:`scorer` for model
+        evaluation. These callables are said to *consume* the sample weights
+        while other components of scikit-learn can *route*  the weights to the
+        underlying estimators or scorers (see
+        :ref:`glossary_metadata_routing`).
+
+        Weighting samples can be useful in several contexts. For instance, if
+        the training data is not uniformly sampled from the target population,
+        it can be corrected by weighting the training data points based on the
+        `inverse probability
+        <https://en.wikipedia.org/wiki/Inverse_probability_weighting>`_ of
+        their selection for training (e.g. inverse propensity weighting). It is
+        also useful to model the frequency of an event of interest per unit of
+        time on a dataset of observations with different exposure durations per
+        individual (see
+        :ref:`sphx_glr_auto_examples_linear_model_plot_poisson_regression_non_normal_loss.py`
+        and
+        :ref:`sphx_glr_auto_examples_linear_model_plot_tweedie_regression_insurance_claims.py`).
+
+        Third-party libraries can also use `sample_weight`-compatible
+        estimators as building blocks to reduce a specific statistical task
+        into a weighted regression or classification task. For instance sample
+        weights can be constructed to adjust a time-to-event model for
+        censoring in a predictive survival analysis setting. In causal
+        inference, it is possible to reduce a conditional average treatment
+        effect estimation task to a weighted regression task under some
+        assumptions. Sample weights can also be used to mitigate
+        fairness-related harms based on a given quantitative definition of
+        fairness.
+
+        Some model hyper-parameters are expressed in terms of a discrete number
+        of samples in a region of the feature space. When fitting with sample
+        weights, a count of samples is often automatically converted to a sum
+        of their weights as is the case  for `min_samples` in
+        :class:`cluster.DBSCAN`, for instance. However, this is not always the
+        case. In particular, the ``min_samples_leaf`` parameter in
+        :class:`ensemble.RandomForestClassifier` does not take weights into
+        account. One should instead pass `min_weight_fraction_leaf` to
+        :class:`ensemble.RandomForestClassifier` to specify the minimum sum of
+        weights of samples in a leaf.
```

Comment:
I think this information is best placed in the docstrings of those classes, not here, or here keep only the first 3-4 lines

Reply:
```suggestion
        Some model hyper-parameters are expressed in terms of a discrete number
        of samples in a region of the feature space. When fitting with sample
        weights, a count of samples is often automatically converted to a sum
        of their weights, but this is not always the case. Please refer to the
        docstring for details.
```